mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java

```diff
@@ -0,0 +1,279 @@
+import static akka.http.javadsl.server.directives.CachingDirectives.alwaysCache;
+
+import akka.actor.ActorSystem;
+import akka.http.caching.javadsl.Cache;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.Uri;
+import akka.http.javadsl.server.PathMatcher0;
+import akka.http.javadsl.server.PathMatchers;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.server.RouteResult;
+import io.mantisrx.master.api.akka.route.Jackson;
+import io.mantisrx.master.api.akka.route.handlers.ResourceClusterRouteHandler;
+import io.mantisrx.master.api.akka.route.v1.HttpRequestMetrics.Endpoints;
+import io.mantisrx.master.api.akka.route.v1.HttpRequestMetrics.HttpVerb;
+import io.mantisrx.master.resourcecluster.proto.GetResourceClusterSpecRequest;
+import io.mantisrx.master.resourcecluster.proto.ListResourceClusterRequest;
+import io.mantisrx.master.resourcecluster.proto.ProvisionResourceClusterRequest;
+import io.mantisrx.master.resourcecluster.proto.ResourceClusterAPIProto.GetResourceClusterResponse;
+import io.mantisrx.master.resourcecluster.proto.ScaleResourceRequest;
+import io.mantisrx.master.resourcecluster.proto.ScaleResourceResponse;
+import io.mantisrx.server.master.config.ConfigurationProvider;
+import io.mantisrx.server.master.config.MasterConfiguration;
+import io.mantisrx.server.master.resourcecluster.ClusterID;
+import io.mantisrx.server.master.resourcecluster.ResourceCluster;
+import io.mantisrx.server.master.resourcecluster.ResourceClusters;
+import io.mantisrx.server.master.resourcecluster.TaskExecutorID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.annotation.VisibleForTesting;
+
+/**
+ * Resource Cluster Route
+ * Defines the following end points:
+ * /api/v1/resourceClusters/list                                      (GET)
+ * <p>
+ * /api/v1/resourceClusters/{}/getResourceOverview                    (GET)
+ * /api/v1/resourceClusters/{}/getRegisteredTaskExecutors             (GET)
+ * /api/v1/resourceClusters/{}/getBusyTaskExecutors                   (GET)
+ * /api/v1/resourceClusters/{}/getAvailableTaskExecutors              (GET)
+ * /api/v1/resourceClusters/{}/getUnregisteredTaskExecutors           (GET)
+ * <p>
+ * /api/v1/resourceClusters/{}/taskExecutors/{}/getTaskExecutorState  (GET)
+ */
+@Slf4j
+//@RequiredArgsConstructor
+public class ResourceClustersNonLeaderRedirectRoute extends BaseRoute {
+
+    private static final PathMatcher0 RESOURCECLUSTERS_API_PREFIX =
+        segment("api").slash("v1").slash("resourceClusters");
+
+    private final ResourceClusters gateway;
+
+    private final ResourceClusterRouteHandler resourceClusterRouteHandler;
+    private final Cache<Uri, RouteResult> routeResultCache;
+
+    // TODO(andyz) integrate w/ masterMain
+    @VisibleForTesting
+    public ResourceClustersNonLeaderRedirectRoute(ResourceClusters gateway) {
```

Comment:
make this package-private if it's only for testing.

Reply:
Removed.